apollo-router/src/query_planner/selection.rs

```diff
@@ -48,97 +48,171 @@ pub(crate) struct InlineFragment {
     pub(crate) selections: Vec<Selection>,
 }
 
-pub(crate) fn select_object(
-    content: &Object,
+pub(crate) fn execute_selection_set<'a>(
```

Comment:
this replaces our initial implementation with a function that closely follows what is happening in the gateway's `executeSelectionSet` function